Source/DafnyLanguageServer/Handlers/DafnyCompletionHandler.cs

```diff
@@ -86,11 +93,81 @@ public CompletionList Process() {
         return new CompletionList();
       }
 
+      private string GetLastTwoCharactersBeforePositionIncluded(TextReader fileContent, DafnyPosition position) {
```

Comment:
I would think this would be a lot simpler if you located the index of the current position and then grabbed the previous two characters. It doesn't seem like processing anything line-by-line buys you anything - "\n@" will still be processed correctly since `\n` is whitespace.